src/Microsoft.DotNet.CodeFormatter.Analyzers/ProvideExplicitVariableTypeAnalyzer.cs

```diff
@@ -30,14 +30,14 @@ public class ProvideExplicitVariableTypeAnalyzer : DiagnosticAnalyzer
                                                                             "Style",
                                                                             DiagnosticSeverity.Warning,
                                                                             true,
-                                                                            customTags: VariableDeclarationCustomTag);
+                                                                            customTags: new[] { VariableDeclarationCustomTag, RuleType.LocalSemantic });
         private static readonly DiagnosticDescriptor s_ruleForEachStatement = new DiagnosticDescriptor(DiagnosticId,
                                                                             ResourceHelper.MakeLocalizableString(nameof(Resources.ExplicitVariableTypeAnalyzer_Title)),
                                                                             ResourceHelper.MakeLocalizableString(nameof(Resources.ExplicitVariableTypeAnalyzer_MessageFormat)),
                                                                             "Style",
                                                                             DiagnosticSeverity.Warning,
                                                                             true,
-                                                                            customTags: ForEachStatementCustomTag);
+                                                                            customTags: new[] { ForEachStatementCustomTag, RuleType.LocalSemantic });
```

Comment:
On a completely unrelated note: is there a plan to upgrade Roslyn reference? I'd like to fix this multiple descriptor issue here by using the new property bag in `Diagnostic`.